packages/features/bookings/Booker/components/hooks/useBookings.ts

```diff
@@ -279,11 +287,21 @@ export const useBookings = ({ event, hashedLink, bookingForm, metadata, teamMemb
       });
     },
     onError: (err, _, ctx) => {
+      window.removeEventListener("beforeunload", handleBeforeUnload);
       // eslint-disable-next-line @calcom/eslint/no-scroll-into-view-embed -- It is only called when user takes an action in embed
       bookerFormErrorRef && bookerFormErrorRef.current?.scrollIntoView({ behavior: "smooth" });
     },
   });
 
+  useEffect(() => {
+    if (!createBookingMutation.isPending) return;
```

Comment:
We should do for recurring one as well I believe. Instant has its own handling, so that should be fine.